### ISSUE TITLE
`destroySession` doesn't delete cookie if max-age is set

### ISSUE THREAD

**Issue by AlexErrant** [remix-run/remix#5150]
### What version of Remix are you using?

v1.11.0

### Steps to Reproduce

1. Create a cookie session storage with `maxAge` set
```ts
  storage = createCookieSessionStorage({
    cookie: {
      name: "__Host-session",
      secure: true,
      sameSite: "lax",
      path: "/",
      maxAge: 60 * 60 * 24 * 30, // 30 days
      httpOnly: true,
    },
  })
```

2. Destroy it
```ts
  const session = await storage.getSession(request.headers.get("Cookie"))
  const headers = new Headers()
  headers.append("Set-Cookie", await storage.destroySession(session))
```

### Expected Behavior

I expected the cookie to be deleted.

### Actual Behavior

Instead, I get a `set-cookie` response header that looks like

`__Host-session=; Max-Age=2592000; Path=/; Expires=Thu, 01 Jan 1970 00:00:00 GMT; HttpOnly; Secure; SameSite=Lax`

Remix blanks out the session, which is great, and causes logouts to work as expected.

Remix also sets the `Expires` set to the epoch, which is a [spec-compliant](https://www.rfc-editor.org/rfc/rfc6265.html#section-4.1.2) way of deleting the cookie:

> servers can delete cookies by sending the user agent a new cookie with an Expires attribute with a value in the past.

However, [the spec](https://www.rfc-editor.org/rfc/rfc6265#section-4.1.2.2) also states:

> If a cookie has both the Max-Age and the Expires attribute, the Max-Age attribute has precedence and controls the expiration date of the cookie.

My interpretation is that if `Max-Age` is set, setting the `Expires` to the epoch doesn't change the cookie's expiration. This is reflected in Chrome, where the above response header does not result in the cookie being deleted.

![Code_YLatP5tpkB](https://user-images.githubusercontent.com/109672176/213527893-fa05a3c7-dff5-486d-8f10-b75ad51ad0e6.png)

May I suggest implementing [this code](https://github.com/remix-run/remix/blob/406029306b77cca37fe4f2bb5de081a22391909b/packages/remix-server-runtime/sessions.ts#L255-L257) as

```ts
        return cookie.serialize("", {
          ...options,
          maxAge: undefined,
          expires: new Date(0)
        });
```

**Comment by AlexErrant** [comment]
Ulgh.

**Comment by mmacdonaldOCP** [comment]
Having this issue as well

**Comment by brophdawg11** [comment]
This is fixed by #7252 and will be available in Remix v2
